### ISSUE TITLE
Go 1.9.4 breaks Solaris release build

### ISSUE THREAD

**Issue by slackpad** [hashicorp/consul#3879]
```
$ make dist

...

--> solaris/amd64 error: exit status 2
Stderr: # github.com/hashicorp/consul/vendor/golang.org/x/sys/unix
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:13:3: //go:cgo_import_dynamic libc_pipe pipe "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:14:3: //go:cgo_import_dynamic libc_getsockname getsockname "libsocket.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:15:3: //go:cgo_import_dynamic libc_getcwd getcwd "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:16:3: //go:cgo_import_dynamic libc_getgroups getgroups "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:17:3: //go:cgo_import_dynamic libc_setgroups setgroups "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:18:3: //go:cgo_import_dynamic libc_wait4 wait4 "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:19:3: //go:cgo_import_dynamic libc_gethostname gethostname "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:20:3: //go:cgo_import_dynamic libc_utimes utimes "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:21:3: //go:cgo_import_dynamic libc_utimensat utimensat "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:22:3: //go:cgo_import_dynamic libc_fcntl fcntl "libc.so" only allowed in cgo-generated code
vendor/golang.org/x/sys/unix/zsyscall_solaris_amd64.go:22:3: too many errors

GNUmakefile:31: recipe for target 'bin' failed
make: *** [bin] Error 1
make: *** [dist] Error 2
```

**Comment by slackpad** [comment]
We reverted back to Go 1.9.3 for tomorrow's release of Consul 1.0.6, but we should investigate this and upgrade Go again as soon as possible.

**Comment by hairyhenderson** [comment]
@slackpad FYI - this was fixed in Go 1.10 - https://golang.org/cl/94018
